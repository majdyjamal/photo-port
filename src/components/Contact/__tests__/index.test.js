import React from 'react'
import { render, cleanup } from '@testing-library/react'
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect'
import ContactForm from '..'

afterEach(cleanup);

describe("ContactForm Component", () => {
    it("Render", () => {
        render(<ContactForm/>);
    });

    it("Match Snapshot", () => {
       const {asFragment} = render(<ContactForm/>);
        // assert value comparison
        expect(asFragment()).toMatchSnapshot();
    });

    // it("Test invalid email", () => {
    //     const {getByTestId, getByText } = render(<ContactForm/>);

    //     userEvent.type( getByTestId("email"), "test@mail.");
    //     getByText('Submit').focus();
    //     expect(getByTestId("errormessage")).toHaveValue("Your email is invalid.");


    // })
});