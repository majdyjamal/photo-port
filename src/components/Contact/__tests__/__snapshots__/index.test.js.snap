// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm Component Match Snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1>
      Contact me
    </h1>
    <form
      id="contact-form"
    >
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email address:
        </label>
        <input
          data-testid="email"
          name="email"
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          name="message"
          rows="5"
        />
      </div>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
